ndb/src/google/cloud/ndb/model.py

```diff
@@ -2538,11 +2538,111 @@ def _now():
         return datetime.datetime.utcnow().time()
 
 
+def _structured_get_for_dict(property_, entity):
+    """Helper for ``_get_for_dict()``.
+
+    This is similar to a mix-in, for :class:`StructuredProperty` and
```

Comment:
The original [used a mix-in][1], but I just felt strange about adding the diamond in the MRO. Happy to re-instate if reviewers would like to.

[1]: https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/cf4cab3f1f69cd04e1a9229871be466b53729f3f/ndb/model.py#L2238